### DISCUSSION TITLE
CTkFrame not obeying height and width when used with place()

### DISCUSSION THREAD

**Discussion by rmills** [TomSchimansky/CustomTkinter#1973] [General]
This may be a user issue as I am new to tkinter, but I can't seem to set a frame size, it just shrinks to whatever is inside and ignores the height and width on the constructor when place() is used. I am stacking frames to make "pages", so I need to explicitly set the frame size. To make things more confusing, in the place() baseclass height and width is a valid kwarg but only to raise the error that they need to be on the constructor? The frame constructor is passing it all the way to base class in tkinter but its just not using it. 

What needs to be called or set for a frame to obey its height and width when its set in the frame constructor?

my example class:

```
import customtkinter as ctk


class BaseFrame(ctk.CTkFrame):
    set_width: int = 1450
    set_height: int = 890
    set_x = 410
    set_y = 140
    menu_btn_width = 340
    menu_btn_height = 50
    menu_btn_font = "DS-Digital Bold"
    menu_btn_font_size = 32
    menu_padding_x = 20
    menu_padding_y = 10

    title_font = "courier"
    title_font_size = 18
    title_padding_x = 20
    title_padding_y = 10

    text_font = "courier"
    text_font_size = 18

    def __init__(self, master, frame_menu: ctk.CTkFrame | None, frame_title: str, **kwargs):
        super().__init__(master, self.set_width, self.set_height, **kwargs)
        self.place(x=self.set_x, y=self.set_y)

        if frame_menu:
            menu_btn = ctk.CTkButton(master=frame_menu, width=self.menu_btn_width, height=self.menu_btn_height, text=frame_title.upper(), font=(self.menu_btn_font, self.menu_btn_font_size), command=self.show)
            menu_btn.pack(padx=self.menu_padding_x, pady=self.menu_padding_y, expand=False)

        title = ctk.CTkLabel(self, text=frame_title, font=(self.title_font, self.title_font_size))
        title.pack(pady=self.title_padding_x, padx=self.title_padding_x, fill="both", expand=True)

    def show(self):
        self.lift()

```

**Comment by DragonOfShuu** [comment]
Generally you don't want to use place, and instead either use pack/grid. In fact, the documentation recommends that you use grid exclusively:

> In this example the `grid` geometry manager is used to set the position and padding of the widget. It is highly recommended to always use the `grid` geometry manager instead of `place` and `pack`, because it is very easy to create responsive and extendable user interfaces.

[link to docs](https://customtkinter.tomschimansky.com/tutorial/grid-system#grid-geometry-manager)

Because of this, I recommend placing all of your frames inside of a grid, and overlap them all on one cell (make sure to set 'sticky' to 'nsew'). You can also, instead of overlapping them all, have a master frame over them that cycles through each one as the user switches pages (you can use `grid_forget()` to unhook one frame, and use `grid()` to hook another).

Hope this helps!
